edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/blob/AzureBlobLogsUploader.cs

```diff
@@ -0,0 +1,155 @@
+            this.iotHubName = Preconditions.CheckNonWhiteSpace(iotHubName, nameof(iotHubName));
+            this.deviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.azureBlobUploader = Preconditions.CheckNotNull(azureBlobUploader, nameof(azureBlobUploader));
+        }
+
+        public async Task Upload(string uri, string id, byte[] payload, LogsContentEncoding logsContentEncoding, LogsContentType logsContentType)
+        {
+            Preconditions.CheckNonWhiteSpace(uri, nameof(uri));
+            Preconditions.CheckNonWhiteSpace(id, nameof(id));
+            Preconditions.CheckNotNull(payload, nameof(payload));
+
+            try
+            {
+                var containerUri = new Uri(uri);
+                string blobName = this.GetBlobName(id, logsContentEncoding, logsContentType);
+                var container = new CloudBlobContainer(containerUri);
+                Events.Uploading(blobName, container.Name);
+                await ExecuteWithRetry(
+                    () =>
+                    {
+                        IAzureBlob blob = this.azureBlobUploader.GetBlob(containerUri, blobName);
+                        SetContentEncoding(blob, logsContentEncoding);
+                        SetContentType(blob, logsContentType);
+                        return blob.UploadFromByteArrayAsync(payload);
+                    },
+                    r => Events.UploadErrorRetrying(blobName, container.Name, r));
+                Events.UploadSuccess(blobName, container.Name);
+            }
+            catch (Exception e)
+            {
+                Events.UploadError(e, id);
+                throw;
+            }
+        }
+
+        internal static string GetExtension(LogsContentEncoding logsContentEncoding, LogsContentType logsContentType)
+        {
+            if (logsContentEncoding == LogsContentEncoding.Gzip)
+            {
+                return "gz";
+            }
+
+            if (logsContentType == LogsContentType.Json)
+            {
+                return "json";
+            }
+
+            return "log";
+        }
+
+        internal string GetBlobName(string id, LogsContentEncoding logsContentEncoding, LogsContentType logsContentType)
+        {
+            string extension = GetExtension(logsContentEncoding, logsContentType);
+            string blobName = $"{id}-{DateTime.UtcNow.ToString("yyyy-MM-dd--HH-mm-ss", CultureInfo.InvariantCulture)}";
+            return $"{this.iotHubName}/{this.deviceId}/{blobName}.{extension}";
+        }
+
+        static void SetContentType(IAzureBlob blob, LogsContentType logsContentType)
+        {
+            switch (logsContentType)
+            {
+                case LogsContentType.Json:
+                    blob.BlobProperties.ContentType = "application/json";
+                    break;
+                case LogsContentType.Text:
+                    blob.BlobProperties.ContentType = "text/plain";
+                    break;
```

Comment:
throw exception in switch default case.

Reply:
Emm, I would rather not - all this does is set the ContentType - if the LogsContentType cannot be parsed, it would be a bit heavy handed to blow the entire operation (than just not setting the BlobProperties.ContentType)